enhancements/mco-kargs-day1-support.md

```diff
@@ -0,0 +1,349 @@
+---
+title: Kernel Arguments - Day 1 Support
+authors:
+  - "@ericavonb"
+reviewers:
+  - "@cgwalters"
+  - "@crawford"
+  - "@imcleod"
+  - "@runcom"
+approvers:
+  - "@cgwalters"
+  - "@crawford"
+  - "@imcleod"
+  - "@runcom"
+
+creation-date: 2019-09-12
+last-updated: 2019-09-23
+status: **provisional**|implementable|implemented|deferred|rejected|withdrawn|replaced
+---
+
+# Kernel Args - Day 1 Support
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Summary
+
+This enhancement proposes adding the ability for admins to configure custom
+kernel arguments during initial RHCOS node configuration. With this enhancement,
+both OS upgrade to the latest [machine-os-content](https://github.com/openshift/machine-config-operator/blob/master/docs/OSUpgrades.md#os-updates)
+and custom kernel args can be applied to a node during the first boot of the
+machine, followed by a single reboot.
+
+## Motivation
+
+Kernel arguments for RHCOS can be set today using a MachineConfig as a
+[`day-2` operation](https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfiguration.md#kernelarguments),
+i.e. after cluster install, incurring an additional reboot to apply. Even if the
+kernel args are configured at [installation time](https://github.com/openshift/installer/blob/master/docs/user/customization.md#install-time-customization-for-machine-configuration),
+the node will still incur a double reboot to apply the args. Furthermore, that
+reboot will not be coordinated by the [MachineConfigController](https://github.com/openshift/machine-config-operator/blob/master/docs/MachineConfigController.md#updatecontroller),
+resulting in possible cluster downtime.
+
+We can provide a better experience by allowing kernel arguments to be set as a
+`day-1` operation, when a node is being initialized with the current OS and
+cluster version.
+
+
+### Goals
+
+- [ ] Cluster admins can define custom kernel arguments for RHCOS nodes during
+  cluster installation, without rebooting the machine twice.
+- [ ] kernel args
+  can be configured via MachineConfigs and will be applied to new nodes joining
+  the cluster, without inclurring an extra reboot.
+- [ ] In clusters installed using the 4.3 installer or higher, nodes with kernel
+  arguments configured via MachineConfigs have these kernel args applied with
+  the update without incurring an extra reboot.
+
+### Non-Goals
+
+- Does not support defining custom kernel arguments on non-RHCOS machines.
+- Does not support applying kernel args on first boot for nodes in clusters
+  installed using a 4.1 or earlier installer.
```

Comment:
Let's fix this to `4.2 or earlier version` because  required karg day1 support will land in 4.3 bootimage